packages/seedless-onboarding-controller/src/SeedlessOnboardingController.ts

```diff
@@ -0,0 +1,836 @@
+  Web3AuthNetwork,
+} from './constants';
+import { RecoveryError } from './errors';
+import { projectLogger, createModuleLogger } from './logger';
+import { SeedPhraseMetadata } from './SeedPhraseMetadata';
+import type {
+  Encryptor,
+  MutuallyExclusiveCallback,
+  SeedlessOnboardingControllerMessenger,
+  SeedlessOnboardingControllerOptions,
+  SeedlessOnboardingControllerState,
+  VaultData,
+} from './types';
+
+const log = createModuleLogger(projectLogger, controllerName);
+
+/**
+ * Seedless Onboarding Controller State Metadata.
+ *
+ * This allows us to choose if fields of the state should be persisted or not
+ * using the `persist` flag; and if they can be sent to Sentry or not, using
+ * the `anonymous` flag.
+ */
+const seedlessOnboardingMetadata: StateMetadata<SeedlessOnboardingControllerState> =
+  {
+    vault: {
+      persist: true,
+      anonymous: false,
+    },
+    backupHashes: {
+      persist: true,
+      anonymous: true,
+    },
+    nodeAuthTokens: {
+      persist: false,
+      anonymous: true,
+    },
+  };
+
+export const defaultState: SeedlessOnboardingControllerState = {
+  backupHashes: [],
+};
+
+export class SeedlessOnboardingController extends BaseController<
+  typeof controllerName,
+  SeedlessOnboardingControllerState,
+  SeedlessOnboardingControllerMessenger
+> {
+  readonly #vaultEncryptor: Encryptor = {
+    encrypt,
+    decrypt,
+  };
+
+  readonly #vaultOperationMutex = new Mutex();
+
+  readonly toprfClient: ToprfSecureBackup;
+
+  constructor({
+    messenger,
+    state,
+    encryptor,
+    network = Web3AuthNetwork.Devnet,
+  }: SeedlessOnboardingControllerOptions) {
+    super({
+      name: controllerName,
+      metadata: seedlessOnboardingMetadata,
+      state: {
+        ...defaultState,
+        ...state,
+      },
+      messenger,
+    });
+
+    if (encryptor) {
+      this.#vaultEncryptor = encryptor;
+    }
+
+    this.toprfClient = new ToprfSecureBackup({
+      network,
+    });
+  }
+
+  /**
+   * @description Authenticate OAuth user using the seedless onboarding flow
+   * and determine if the user is already registered or not.
+   * @param params - The parameters for authenticate OAuth user.
+   * @param params.idTokens - The ID token from Social login
+   * @param params.authConnectionId - OAuth authConnectionId from dashboard
+   * @param params.userId - user email or id from Social login
+   * @param params.groupedAuthConnectionId - Optional grouped authConnectionId to be used for the authenticate request.
+   * You can pass this to use aggregate connection.
+   * @returns A promise that resolves to the authentication result.
+   */
+  async authenticate(params: {
+    idTokens: string[];
+    authConnectionId: string;
+    groupedAuthConnectionId?: string;
+    userId: string;
+  }) {
+    try {
+      const { idTokens, authConnectionId, groupedAuthConnectionId, userId } =
+        params;
+      const verifier = groupedAuthConnectionId || authConnectionId;
```

Comment:
we should use same new terminology consistently for connection and user ids, rather thn using verifier and verifierIds